Godot 4.2.1 Compatibility Rendering (GLES3) don't works at all on a 3 Years Old Andoid Device

### Tested versions

Godot 4.2.1

### System information

Android 8.0 - Godot 4.2.1 Compatibility Rendering (GLES3)

### Issue description

I created an empty project with an empty scene with the Compatibility Rendering modality, after exported it for Android and upload to my Old Huawei 10" Tablet (Android 8 updated on Sep/2020), results?
After showing the Godot Logo the app freeze.

Initially, I was discouraged thinking that my old tablet probably missing the OpenGL ES 3.0 driver/feature (GLES3) but to be sure I installed GFXBench, an Android App able to test and bench all the device’s graphic abilities.
The app showed me that my device has full support for OpenGL ES 1.0, 2.0, and 3.0 but hasn’t any support for Vulkan, OpenGL ES 3.1 and 3.2.
To be sure I executed all the OpenGL ES 3.0 tests and to my great surprise, all works fine, as you can see from the following image.
![Screenshot_20231213-104133](https://github.com/godotengine/godot/assets/19651914/0fc21859-bf7a-48c7-931e-8239e612d4c2)

So now my question is… how can I help to fix this bug?

### Steps to reproduce

You need an Old Android Device.

### Minimal reproduction project (MRP)

Any empty project with an empty scene in Compatibility Modality

> We need a bit more information from @joined72:
> 
>     1. Have you tested GLES3 in Godot 3.5.2? It uses more advanced features than the compatibility renderer in Godot 4.2, so if it works, then 4.2 should definitely work

I just tested the game on Godot 3.5.3 GLES3, ... same results! :(
 
>     2. Can you share your project, or at least your project.godot file so we can confirm that you are exporting with OpenGL?

Here: [GLES3_Test.zip](https://github.com/godotengine/godot/files/13671160/GLES3_Test.zip)

 
>     3. Have you tried debugging on your device using `adb logcat` to get the logs? It would be really helpful for us to see if the engine is printing any error messages as it loads

This is the `adb logcat` obtained with the `adb logcat | grep -E 'godot|com.example.gles3_test'` command: 

```
12-13 11:13:22.600 24048 24048 E AndroidRuntime: Process: org.godotengine.godot4gles3, PID: 24048
12-13 11:13:22.600 24048 24048 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{org.godotengine.godot4gles3/com.godot.game.GodotApp}: java.lang.IllegalStateException: Unable to initialize engine render view
12-13 11:13:22.600 24048 24048 E AndroidRuntime: 	at org.godotengine.godot.GodotFragment.performEngineInitialization(GodotFragment.java:196)
12-13 11:13:22.600 24048 24048 E AndroidRuntime: 	at org.godotengine.godot.GodotFragment.onCreate(GodotFragment.java:182)
12-13 11:13:22.600 24048 24048 E AndroidRuntime: 	at org.godotengine.godot.GodotActivity.onCreate(GodotActivity.kt:84)
12-13 11:13:22.600 24048 24048 E AndroidRuntime: 	at com.godot.game.GodotApp.onCreate(GodotApp.java:45)
12-14 09:14:17.589   461   465 D VoldCmdListener: volume mkdirs /storage/emulated/0/Android/data/org.godotengine.godot3gles3/files/
12-14 09:14:18.059 20163 20163 D ActivityThread: add activity client record, r= ActivityRecord{1040240 token=android.os.BinderProxy@4ed6129 {org.godotengine.godot3gles3/com.godot.game.GodotApp}} token= android.os.BinderProxy@4ed6129
12-14 09:15:42.981 20542 20542 D ActivityThread: add activity client record, r= ActivityRecord{1040240 token=android.os.BinderProxy@4ed6129 {org.godotengine.godot3gles3/com.godot.game.GodotApp}} token= android.os.BinderProxy@4ed6129
12-14 09:26:17.462   461   465 D VoldCmdListener: volume mkdirs /storage/emulated/0/Android/data/com.example.gles3_test/files/
12-14 09:26:17.881 22055 22055 D ActivityThread: add activity client record, r= ActivityRecord{1040240 token=android.os.BinderProxy@4ed6129 {com.example.gles3_test/com.godot.game.GodotApp}} token= android.os.BinderProxy@4ed6129
12-14 09:38:41.111 24447 24447 D ActivityThread: add activity client record, r= ActivityRecord{1040240 token=android.os.BinderProxy@4ed6129 {com.example.gles3_test/com.godot.game.GodotApp}} token= android.os.BinderProxy@4ed6129
12-14 09:41:56.174 24878 24878 D ActivityThread: add activity client record, r= ActivityRecord{1040240 token=android.os.BinderProxy@4ed6129 {com.example.gles3_test/com.godot.game.GodotApp}} token= android.os.BinderProxy@4ed6129
12-14 09:42:32.547 25000 25000 D ActivityThread: add activity client record, r= ActivityRecord{1040240 token=android.os.BinderProxy@4ed6129 {com.example.gles3_test/com.godot.game.GodotApp}} token= android.os.BinderProxy@4ed6129
12-14 09:57:20.675   777  1719 D installed: forkNotifyToRMS parcel pid = 26781, ppid = 777, pkgname = com.example.gles3_test
12-14 09:57:20.824 26781 26781 I dex2oat : /system/bin/dex2oat Installd-com.example.gles3_test --input-vdex-fd=-1 --output-vdex-fd=21 --compiler-filter=quicken --debuggable --classpath-dir=/data/app/vmdl2016444544.tmp
12-14 09:57:22.867 14845 26793 W GCM     : Unexpected forwarded intent: Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.example.gles3_test flg=0x4000010 pkg=com.google.android.gms cmp=com.google.android.gms/.chimera.PersistentIntentOperationService (has extras) }
12-14 09:57:22.886 14845 26795 W GCM     : Unexpected forwarded intent: Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.example.gles3_test flg=0x4000010 pkg=com.google.android.gms cmp=com.google.android.gms/.chimera.PersistentIntentOperationService (has extras) }
12-14 09:57:23.262 22020 22142 D WorkSource: i1=1 i2=0 N1=1: insert 10109 com.example.gles3_test
12-14 09:57:23.263 22020 22142 D WorkSource: Insert in WorkSource{10085 com.google.android.gms} @ 1 uid 10109 name com.example.gles3_test
12-14 09:57:23.271 24641 24691 I Finsky  : [2593] knt.a(55): AIM: AppInfoManager-Perf > OnDeviceAppInfo > cacheHitCount=0, cacheMissCount=1. Missed  in cache (limit 10) : [com.example.gles3_test]
12-14 09:57:23.304 22020 23844 D WorkSource: Insert in WorkSource{10109 com.example.gles3_test} @ 0 uid 10085 name com.google.android.gms
12-14 09:57:23.670  2271  2271 I HwLauncher: Model onPackageAdded packageName = com.example.gles3_test user = UserHandle{0}
12-14 09:57:23.691  2271  2271 I HwLauncher: Model  findCloudAppInfoInDB : found 0 shortcutInfo for package[com.example.gles3_test] in db
12-14 09:57:23.691  2271  2271 I HwLauncher: PowerSaveLauncherModel onPackageAdded packageName = com.example.gles3_test user = UserHandle{0}
12-14 09:57:23.691  2271  2271 I HwLauncher: StreetModeLauncherModel onPackageAdded packageName = com.example.gles3_test user = UserHandle{0}
12-14 09:57:23.936 24641 24641 I Finsky  : [1] kly.f(9): AIM: AppInfoCacheUpdater -> invalidating apps: [com.example.gles3_test]
12-14 09:57:23.950 24641 24691 I Finsky  : [2593] knt.a(55): AIM: AppInfoManager-Perf > OnDeviceAppInfo > cacheHitCount=0, cacheMissCount=1. Missed  in cache (limit 10) : [com.example.gles3_test]
12-14 09:57:24.115 24641 26450 I Finsky  : [2674] jua.a(178): Asset module storage cleared for package com.example.gles3_test.
12-14 09:57:24.489 14845 26801 W GCM     : Unexpected forwarded intent: Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.example.gles3_test flg=0x4000010 pkg=com.google.android.gms cmp=com.google.android.gms/.chimera.PersistentIntentOperationService (has extras) }
12-14 09:57:24.491 14845 26795 W GCM     : Unexpected forwarded intent: Intent { act=android.intent.action.PACKAGE_ADDED dat=package:com.example.gles3_test flg=0x4000010 pkg=com.google.android.gms cmp=com.google.android.gms/.chimera.PersistentIntentOperationService (has extras) }
12-14 09:57:24.610 22020 22142 D WorkSource: i1=1 i2=0 N1=1: insert 10109 com.example.gles3_test
12-14 09:57:24.610 22020 22142 D WorkSource: Insert in WorkSource{10085 com.google.android.gms} @ 1 uid 10109 name com.example.gles3_test
12-14 09:59:46.699  2219  2606 I ash     : front pkg : com.example.gles3_test launcher: false
12-14 09:59:46.699  2219  2606 I ash     : above launcher front pkgs: [com.example.gles3_test]
12-14 09:59:47.020 27048 27048 I chatty  : uid=10125(u0_a125) com.example.gles3_test identical 345 lines
12-14 09:59:47.134  2219  2357 D ScenarioService: front pkg : com.example.gles3_test launcher: false
12-14 09:59:47.134  2219  2357 D ScenarioService: new above launcher front pkgs: [com.example.gles3_test]
12-14 09:59:47.139  2219  2357 I AppsUsage: scnOff:false FgAPP:com.example.gles3_test BgAPP:com.huawei.android.launcher
12-14 09:59:47.176  2219  2798 I PGServer: report state:10000 event type:1 pid:0 uid:0 pkg:com.example.gles3_test to pid: 1718
12-14 09:59:47.176  2219  2798 I PGServer: report state:10000 event type:1 pid:0 uid:0 pkg:com.example.gles3_test to pid: 2219
12-14 09:59:47.176  2219  2798 I SceneReceiver: state type: 10000 eventType:1 pid:0 uid:0 pkg:com.example.gles3_test
12-14 09:59:47.177  2219  2812 D HwThermalStateManager: current foreapp pkg : com.example.gles3_test
12-14 09:59:47.274 27048 27064 I chatty  : uid=10125(u0_a125) com.example.gles3_test expire 14 lines
12-14 09:59:47.369   461   465 D VoldCmdListener: volume mkdirs /storage/emulated/0/Android/data/com.example.gles3_test/files/
12-14 09:59:47.765 27048 27048 D ActivityThread: add activity client record, r= ActivityRecord{1040240 token=android.os.BinderProxy@4ed6129 {com.example.gles3_test/com.godot.game.GodotApp}} token= android.os.BinderProxy@4ed6129
12-14 09:59:47.819 27048 27079 I chatty  : uid=10125(u0_a125) com.example.gles3_test expire 7 lines
12-14 09:59:47.843 27048 27080 I chatty  : uid=10125(u0_a125) com.example.gles3_test expire 1 line
12-14 09:59:47.950 27048 27077 I chatty  : uid=10125(u0_a125) com.example.gles3_test expire 18 lines
12-14 09:59:54.242 27048 27054 I chatty  : uid=10125(u0_a125) com.example.gles3_test expire 3 lines
12-14 10:00:22.265 27048 27048 I chatty  : uid=10125(u0_a125) com.example.gles3_test expire 1 line
12-14 10:00:25.360  2219  2357 D ScenarioService: new above launcher front pkgs: [com.example.gles3_test, com.android.systemui]
12-14 10:00:25.360  2219  2357 I AppsUsage: scnOff:false FgAPP:com.android.systemui BgAPP:com.example.gles3_test
12-14 10:00:25.422  2219  2606 I GpsController: com.example.gles3_test is above launcher app, just return...
12-14 10:00:25.422  2219  2606 I ash     : above launcher front pkgs: [com.example.gles3_test, com.android.systemui]
12-14 10:00:25.424  2219  2798 I PGServer: report state:10000 event type:2 pid:0 uid:0 pkg:com.example.gles3_test to pid: 1718
12-14 10:00:25.425  2219  2798 I PGServer: report state:10000 event type:2 pid:0 uid:0 pkg:com.example.gles3_test to pid: 2219
12-14 10:00:25.425  2219  2798 I SceneReceiver: state type: 10000 eventType:2 pid:0 uid:0 pkg:com.example.gles3_test
12-14 10:00:58.825 27215 27215 I chatty  : uid=10125(u0_a125) com.example.gles3_test identical 39 lines
12-14 10:00:59.046 27215 27232 I chatty  : uid=10125(u0_a125) com.example.gles3_test expire 13 lines
12-14 10:00:59.195 27215 27215 I zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.WindowInsetsAnimation$Callback" on path: DexPathList[[zip file "/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk"],nativeLibraryDirectories=[/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/lib/arm64, /data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64, /product/lib64]]
12-14 10:00:59.195 27215 27215 I zygote64:   at void org.godotengine.godot.GodotFragment.onCreate(android.os.Bundle) (GodotFragment.java:181)
12-14 10:00:59.196 27215 27215 I zygote64:   at void org.godotengine.godot.GodotActivity.onCreate(android.os.Bundle) (GodotActivity.kt:84)
12-14 10:00:59.196 27215 27215 I zygote64:   at void com.godot.game.GodotApp.onCreate(android.os.Bundle) (GodotApp.java:45)
12-14 10:00:59.196 27215 27215 I zygote64: Rejecting re-init on previously-failed class java.lang.Class<org.godotengine.godot.Godot$onInitRenderView$2>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/WindowInsetsAnimation$Callback;
12-14 10:00:59.197 27215 27215 I zygote64:   at void org.godotengine.godot.GodotFragment.onCreate(android.os.Bundle) (GodotFragment.java:181)
12-14 10:00:59.197 27215 27215 I zygote64:   at void org.godotengine.godot.GodotActivity.onCreate(android.os.Bundle) (GodotActivity.kt:84)
12-14 10:00:59.197 27215 27215 I zygote64:   at void com.godot.game.GodotApp.onCreate(android.os.Bundle) (GodotApp.java:45)
12-14 10:00:59.197 27215 27215 I zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.WindowInsetsAnimation$Callback" on path: DexPathList[[zip file "/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk"],nativeLibraryDirectories=[/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/lib/arm64, /data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64, /product/lib64]]
12-14 10:00:59.197 27215 27215 I zygote64:   at void org.godotengine.godot.GodotFragment.onCreate(android.os.Bundle) (GodotFragment.java:181)
12-14 10:00:59.197 27215 27215 I zygote64:   at void org.godotengine.godot.GodotActivity.onCreate(android.os.Bundle) (GodotActivity.kt:84)
12-14 10:00:59.197 27215 27215 I zygote64:   at void com.godot.game.GodotApp.onCreate(android.os.Bundle) (GodotApp.java:45)
12-14 10:00:59.198 27215 27215 I zygote64: Rejecting re-init on previously-failed class java.lang.Class<org.godotengine.godot.Godot$onInitRenderView$2>: java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/WindowInsetsAnimation$Callback;
12-14 10:00:59.198 27215 27215 I zygote64:   at void org.godotengine.godot.GodotFragment.onCreate(android.os.Bundle) (GodotFragment.java:181)
12-14 10:00:59.198 27215 27215 I zygote64:   at void org.godotengine.godot.GodotActivity.onCreate(android.os.Bundle) (GodotActivity.kt:84)
12-14 10:00:59.198 27215 27215 I zygote64:   at void com.godot.game.GodotApp.onCreate(android.os.Bundle) (GodotApp.java:45)
12-14 10:00:59.198 27215 27215 I zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.WindowInsetsAnimation$Callback" on path: DexPathList[[zip file "/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk"],nativeLibraryDirectories=[/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/lib/arm64, /data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64, /product/lib64]]
12-14 10:00:59.198 27215 27215 I zygote64:   at void org.godotengine.godot.GodotFragment.onCreate(android.os.Bundle) (GodotFragment.java:181)
12-14 10:00:59.198 27215 27215 I zygote64:   at void org.godotengine.godot.GodotActivity.onCreate(android.os.Bundle) (GodotActivity.kt:84)
12-14 10:00:59.198 27215 27215 I zygote64:   at void com.godot.game.GodotApp.onCreate(android.os.Bundle) (GodotApp.java:45)
12-14 10:00:59.218 27215 27215 I HwApiCacheMangerEx: apicache pi PackageInfo{e4d93c8 com.example.gles3_test}
12-14 10:00:59.230 27215 27215 W linker  : "/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/lib/arm64/libc++_shared.so" unused DT entry: type 0x70000001 arg 0x0
12-14 10:00:59.396 27215 27215 I zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk"],nativeLibraryDirectories=[/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/lib/arm64, /data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64, /product/lib64]]
12-14 10:00:59.397 27215 27215 I zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk"],nativeLibraryDirectories=[/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/lib/arm64, /data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64, /product/lib64]]
12-14 10:00:59.398 27215 27215 I zygote64: Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.View$OnUnhandledKeyEventListener" on path: DexPathList[[zip file "/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk"],nativeLibraryDirectories=[/data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/lib/arm64, /data/app/com.example.gles3_test-fsGPOybXNVecQhPMFjbE7A==/base.apk!/lib/arm64-v8a, /system/lib64, /vendor/lib64, /product/lib64]]
12-14 10:00:59.426 27215 27215 D ActivityThread: add activity client record, r= ActivityRecord{1040240 token=android.os.BinderProxy@4ed6129 {com.example.gles3_test/com.godot.game.GodotApp}} token= android.os.BinderProxy@4ed6129
12-14 10:00:59.565 27215 27215 W com.example.gles3_test: type=1400 audit(0.0:2273): avc: denied { search } for comm=474C5468726561642032363430 name="kgsl-3d0" dev="sysfs" ino=21562 scontext=u:r:untrusted_app:s0:c512,c768 tcontext=u:object_r:sysfs_kgsl:s0 tclass=dir permissive=0
12-14 10:00:59.579 27215 27244 I godot   : Godot Engine v4.2.1.stable.official.b09f793f5 - https://godotengine.org
12-14 10:00:59.604 27215 27244 I godot   : TextServer: Added interface "Dummy"
12-14 10:00:59.690 27215 27244 I godot   : TextServer: Added interface "ICU / HarfBuzz / Graphite (Built-in)"
12-14 10:00:59.726 27215 27244 I godot   : Using "default" pen tablet driver...
12-14 10:00:59.729 27215 27244 I godot   : godot: ENABLING GL DEBUG
12-14 10:00:59.738 27215 27244 I godot   : Shader 'CanvasSdfShaderGLES3' SHA256: a86e8644c91faf628a28d82fbb43d8062537e54756e49ba358effb980bee9625
12-14 10:00:59.747 27215 27244 I godot   : Shader 'SkeletonShaderGLES3' SHA256: b7f963aa549ccea4474fe7cbd3ff1314d5b592f9d25187a0753390901abdeebf
12-14 10:00:59.750 27215 27244 I godot   : Shader 'ParticlesShaderGLES3' SHA256: fa0ed3a7b23c664eafb1ad09a42372cae71d497e726bcaac074a1591e5e2ffbd
12-14 10:00:59.752 27215 27244 I godot   : Shader 'ParticlesCopyShaderGLES3' SHA256: 80fe86edb00f96ba8389672e2268cecc06c577dbe1a191b13204c0bb6b30fcb6
12-14 10:00:59.753 27215 27244 I godot   : Shader 'CopyShaderGLES3' SHA256: a6d403c8b183ba79926a177b814b5c2c0e0427713ba49b70370c94012591dc41
12-14 10:00:59.763 27215 27244 I godot   : Shader 'CanvasShaderGLES3' SHA256: a77bbb6fcf4720cba41376e989b731b63a27bceca4ea7f371210607385b49674
12-14 10:00:59.764 27215 27244 I godot   : Shader 'CanvasOcclusionShaderGLES3' SHA256: bf78a03bdac6d21135277216f7c9cb5eb789dbb9da1eec409014512904514cba
12-14 10:00:59.778 27215 27244 I godot   : Shader 'SceneShaderGLES3' SHA256: 1783396bc13d69aba84a33963083b18661a6e26af5b4b4dbb5e09b4b97f368cc
12-14 10:02:59.989  1718  1881 V WindowManager: removeIfPossible: Window{41162b5 u0 Splash Screen com.example.gles3_test} callers=com.android.server.wm.WindowState.removeIfPossible:1914 com.android.server.wm.WindowManagerService.removeWindow:2045 com.android.server.wm.Session.remove:218 android.view.ViewRootImpl.dispatchDetachedFromWindow:3774 android.view.ViewRootImpl.doDie:6861 
12-14 10:02:59.990  1718  1881 I WindowManager: Destroying surface Surface(name=Splash Screen com.example.gles3_test) called by com.android.server.wm.WindowStateAnimator.destroySurface:2092 com.android.server.wm.WindowStateAnimator.destroySurfaceLocked:899 com.android.server.wm.HwWindowStateAnimator.destroySurfaceLocked:1006 com.android.server.wm.WindowState.removeImmediately:1899 com.android.server.wm.WindowState.removeIfPossible:2051 com.android.server.wm.WindowState.removeIfPossible:1914 com.android.server.wm.WindowManagerService.removeWindow:2045 com.android.server.wm.Session.remove:218 
12-14 10:02:59.991  1718  1881 V WindowManager: postWindowRemoveCleanupLocked: Window{41162b5 u0 Splash Screen com.example.gles3_test}
12-14 10:02:59.991  1718  1881 V WindowManager: Removing Window{41162b5 u0 Splash Screen com.example.gles3_test} from AppWindowToken{8517aa8 token=Token{39d42cb ActivityRecord{486599a u0 com.example.gles3_test/com.godot.game.GodotApp t34}}}
12-14 10:02:59.998  1718  3971 W HwActivityManagerServiceEx: appSwitch from: com.example.gles3_test to: com.android.systemui
12-14 10:02:59.999  1718  3971 V WindowManager: findFocusedWindow: AppWindowToken{8517aa8 token=Token{39d42cb ActivityRecord{486599a u0 com.example.gles3_test/com.godot.game.GodotApp t34}}} below Reached focused app=AppWindowToken{4334448 token=Token{c67aeeb ActivityRecord{c4623a u0 com.android.systemui/.recents.RecentsActivity t13}}}
12-14 10:03:00.017  1718  3971 I WindowManager: Failed to capture screenshot of Token{39d42cb ActivityRecord{486599a u0 com.example.gles3_test/com.godot.game.GodotApp t34}} appWin=Window{732c033 u0 com.example.gles3_test/com.godot.game.GodotApp} drawState=1
12-14 10:03:00.247   791  1022 I dubaid  : [AbnormalityRoutine.cpp] checkCpuAbnormality# 10108:org.godotengine.godot3gles3:0.249:380710:0
12-14 10:03:00.498  1718  1799 W ActivityManager: Activity pause timeout for ActivityRecord{486599a u0 com.example.gles3_test/com.godot.game.GodotApp t34}
12-14 10:03:00.603  1718  2164 D ActivityManager: Skipping intent:act=android.intent.action.DELETE dat=package:com.example.gles3_test flg=0x10800000 cmp=com.google.android.packageinstaller/com.android.packageinstaller.UninstallerActivity
12-14 10:03:01.147  2030  2136 D ReporterUtil:  e id:38 type:{pkg:com.example.gles3_test}
12-14 10:03:01.371  1718 27412 I ActivityManager: Killing 27215:com.example.gles3_test/u0a125 (adj 900): remove task
12-14 10:03:01.401  1718  3355 I WindowManager: WIN DEATH: Window{732c033 u0 com.example.gles3_test/com.godot.game.GodotApp}
12-14 10:03:01.401  1718  3355 V WindowManager: removeIfPossible: Window{732c033 u0 com.example.gles3_test/com.godot.game.GodotApp} callers=com.android.server.wm.WindowState.-wrap1:0 com.android.server.wm.WindowState$DeathRecipient.binderDied:2481 android.os.BinderProxy.sendDeathNotice:842 <bottom of call stack> <bottom of call stack> 
12-14 10:03:01.402  1718  3355 I WindowManager: Destroying surface Surface(name=com.example.gles3_test/com.godot.game.GodotApp) called by com.android.server.wm.WindowStateAnimator.destroySurface:2092 com.android.server.wm.WindowStateAnimator.destroySurfaceLocked:899 com.android.server.wm.HwWindowStateAnimator.destroySurfaceLocked:1006 com.android.server.wm.WindowState.removeImmediately:1899 com.android.server.wm.WindowState.removeIfPossible:2051 com.android.server.wm.WindowState.-wrap1:0 com.android.server.wm.WindowState$DeathRecipient.binderDied:2481 android.os
```
The second log line seems to indicate a problem in initializing the GLES3 render.

`E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{org.godotengine.godot4gles3/com.godot.game.GodotApp}: java.lang.IllegalStateException: Unable to initialize engine render view`

--- a/drivers/gles3/rasterizer_gles3.cpp
+++ b/drivers/gles3/rasterizer_gles3.cpp
@@ -194,6 +194,11 @@ typedef void(GLAPIENTRY *DebugMessageCallbackARB)(DEBUGPROCARB callback, const v
 
 void RasterizerGLES3::initialize() {
 	Engine::get_singleton()->print_header(vformat("OpenGL API %s - Compatibility - Using Device: %s - %s", RS::get_singleton()->get_video_adapter_api_version(), RS::get_singleton()->get_video_adapter_vendor(), RS::get_singleton()->get_video_adapter_name()));
+
+	// FLIP XY Bug: Are more devices affected?
+	// Confirmed so far: all Adreno 3xx
+	// ok on some tested Adreno devices: 4xx, 5xx and 6xx
+	flip_xy_bugfix = GLES3::Config::get_singleton()->adreno_3xx_compatibility;
 }
 
 void RasterizerGLES3::finalize() {
@@ -398,8 +403,18 @@ void RasterizerGLES3::_blit_render_target_to_screen(RID p_render_target, Display
 	}
 
 	Vector2i screen_rect_end = p_screen_rect.get_end();
+
+	// Adreno (TM) 3xx devices have a bug that create wrong Landscape rotation of 180 degree
+	// Reversing both the X and Y axis is equivalent to rotating 180 degrees
+	bool flip_x = false;
+	if (flip_xy_bugfix && screen_rect_end.x > screen_rect_end.y) {
+		flip_y = !flip_y;
+		flip_x = !flip_x;
+	}
+
 	glBlitFramebuffer(0, 0, rt->size.x, rt->size.y,
-			p_screen_rect.position.x, flip_y ? screen_rect_end.y : p_screen_rect.position.y, screen_rect_end.x, flip_y ? p_screen_rect.position.y : screen_rect_end.y,
+			flip_x ? screen_rect_end.x : p_screen_rect.position.x, flip_y ? screen_rect_end.y : p_screen_rect.position.y,
+			flip_x ? p_screen_rect.position.x : screen_rect_end.x, flip_y ? p_screen_rect.position.y : screen_rect_end.y,
 			GL_COLOR_BUFFER_BIT, GL_NEAREST);
 
 	if (read_fbo != 0) {

--- a/drivers/gles3/rasterizer_gles3.h
+++ b/drivers/gles3/rasterizer_gles3.h
@@ -55,6 +55,7 @@ private:
 	float delta = 0;
 
 	double time_total = 0.0;
+	bool flip_xy_bugfix = false;
 
 	static bool gles_over_gl;
 

--- a/drivers/gles3/shaders/canvas.glsl
+++ b/drivers/gles3/shaders/canvas.glsl
@@ -187,8 +187,31 @@ void main() {
 #endif // !USE_INSTANCING
 
 #else // !USE_ATTRIBUTES
-	vec2 vertex_base_arr[6] = vec2[](vec2(0.0, 0.0), vec2(0.0, 1.0), vec2(1.0, 1.0), vec2(1.0, 0.0), vec2(0.0, 0.0), vec2(1.0, 1.0));
-	vec2 vertex_base = vertex_base_arr[gl_VertexID % 6];
+
+	// crash on Adreno 320/330
+	//vec2 vertex_base_arr[6] = vec2[](vec2(0.0, 0.0), vec2(0.0, 1.0), vec2(1.0, 1.0), vec2(1.0, 0.0), vec2(0.0, 0.0), vec2(1.0, 1.0));
+	//vec2 vertex_base = vertex_base_arr[gl_VertexID % 6];
+	//-----------------------------------------
+	// ID |  0  |  1  |  2  |  3  |  4  |  5  |
+	//-----------------------------------------
+	// X  | 0.0 | 0.0 | 1.0 | 1.0 | 0.0 | 1.0 |
+	// Y  | 0.0 | 1.0 | 1.0 | 0.0 | 0.0 | 1.0 |
+	//-----------------------------------------
+	// no crash or freeze on all Adreno 3xx	with 'if / else if' and slightly faster!
+	int vertex_id = gl_VertexID % 6;
+	vec2 vertex_base;
+	if (vertex_id == 0)
+		vertex_base = vec2(0.0, 0.0);
+	else if (vertex_id == 1)
+		vertex_base = vec2(0.0, 1.0);
+	else if (vertex_id == 2)
+		vertex_base = vec2(1.0, 1.0);
+	else if (vertex_id == 3)
+		vertex_base = vec2(1.0, 0.0);
+	else if (vertex_id == 4)
+		vertex_base = vec2(0.0, 0.0);
+	else if (vertex_id == 5)
+		vertex_base = vec2(1.0, 1.0);
 
 	vec2 uv = read_draw_data_src_rect.xy + abs(read_draw_data_src_rect.zw) * ((read_draw_data_flags & FLAGS_TRANSPOSE_RECT) != uint(0) ? vertex_base.yx : vertex_base.xy);
 	vec4 color = read_draw_data_modulation;
@@ -475,16 +498,12 @@ vec4 light_shadow_compute(uint light_base, vec4 light_color, vec4 shadow_uv
 void light_blend_compute(uint light_base, vec4 light_color, inout vec3 color) {
 	uint blend_mode = light_array[light_base].flags & LIGHT_FLAGS_BLEND_MASK;
 
-	switch (blend_mode) {
-		case LIGHT_FLAGS_BLEND_MODE_ADD: {
-			color.rgb += light_color.rgb * light_color.a;
-		} break;
-		case LIGHT_FLAGS_BLEND_MODE_SUB: {
-			color.rgb -= light_color.rgb * light_color.a;
-		} break;
-		case LIGHT_FLAGS_BLEND_MODE_MIX: {
-			color.rgb = mix(color.rgb, light_color.rgb, light_color.a);
-		} break;
+	if (blend_mode == LIGHT_FLAGS_BLEND_MODE_ADD) {
+		color.rgb += light_color.rgb * light_color.a;
+	} else if (blend_mode == LIGHT_FLAGS_BLEND_MODE_SUB) {
+		color.rgb -= light_color.rgb * light_color.a;
+	} else if (blend_mode == LIGHT_FLAGS_BLEND_MODE_MIX) {
+		color.rgb = mix(color.rgb, light_color.rgb, light_color.a);
 	}
 }
 

--- a/drivers/gles3/shaders/scene.glsl
+++ b/drivers/gles3/shaders/scene.glsl
@@ -1200,7 +1200,10 @@ void light_process_spot(uint idx, vec3 vertex, vec3 eye_vec, vec3 normal, vec3 f
 	vec3 spot_dir = spot_lights[idx].direction;
 	float scos = max(dot(-normalize(light_rel_vec), spot_dir), spot_lights[idx].cone_angle);
 	float spot_rim = max(0.0001, (1.0 - scos) / (1.0 - spot_lights[idx].cone_angle));
-	spot_attenuation *= 1.0 - pow(spot_rim, spot_lights[idx].cone_attenuation);
+
+	mediump float cone_attenuation = spot_lights[idx].cone_attenuation;
+	spot_attenuation *= 1.0 - pow(spot_rim, cone_attenuation);
+
 	vec3 color = spot_lights[idx].color;
 
 	float size_A = 0.0;

--- a/drivers/gles3/storage/config.cpp
+++ b/drivers/gles3/storage/config.cpp
@@ -166,6 +166,11 @@ Config::Config() {
 	max_renderable_elements = GLOBAL_GET("rendering/limits/opengl/max_renderable_elements");
 	max_renderable_lights = GLOBAL_GET("rendering/limits/opengl/max_renderable_lights");
 	max_lights_per_object = GLOBAL_GET("rendering/limits/opengl/max_lights_per_object");
+
+	//Adreno 3xx Compatibility
+	const String rendering_device_name = String::utf8((const char *)glGetString(GL_RENDERER));
+	//TODO: Check the number between 300 and 399(?)
+	adreno_3xx_compatibility = (rendering_device_name.left(13) == "Adreno (TM) 3");
 }
 
 Config::~Config() {

--- a/drivers/gles3/storage/config.h
+++ b/drivers/gles3/storage/config.h
@@ -91,6 +91,8 @@ public:
 	bool rt_msaa_multiview_supported = false;
 	bool multiview_supported = false;
 
+	bool adreno_3xx_compatibility = false;
+
 #ifdef ANDROID_ENABLED
 	PFNGLFRAMEBUFFERTEXTUREMULTIVIEWOVRPROC eglFramebufferTextureMultiviewOVR = nullptr;
 	PFNGLTEXSTORAGE3DMULTISAMPLEPROC eglTexStorage3DMultisample = nullptr;

--- a/platform/android/java/lib/src/org/godotengine/godot/gl/GLSurfaceView.java
+++ b/platform/android/java/lib/src/org/godotengine/godot/gl/GLSurfaceView.java
@@ -1673,7 +1673,24 @@ public class GLSurfaceView extends SurfaceView implements SurfaceHolder.Callback
 				mWantRenderNotification = true;
 				mRequestRender = true;
 				mRenderComplete = false;
-				mFinishDrawingRunnable = finishDrawing;
+
+				// fix lost old callback when continuous call requestRenderAndNotify
+				//
+				// If continuous call requestRenderAndNotify before trigger old
+				// callback, old callback will lose, cause VRI will wait for SV's
+				// draw to finish forever not calling finishDraw.
+				// https://android.googlesource.com/platform/frameworks/base/+/044fce0b826f2da3a192aac56785b5089143e693%5E%21/
+				//+++++++++++++++++++++++++++++++++++++++++++++++++++
+				final Runnable oldCallback = mFinishDrawingRunnable;
+				mFinishDrawingRunnable = () -> {
+					if (oldCallback != null) {
+						oldCallback.run();
+					}
+					if (finishDrawing != null) {
+						finishDrawing.run();
+					}
+				};
+				//----------------------------------------------------
 
 				sGLThreadManager.notifyAll();
 			}

--- a/platform/android/java/lib/src/org/godotengine/godot/xr/regular/RegularContextFactory.java
+++ b/platform/android/java/lib/src/org/godotengine/godot/xr/regular/RegularContextFactory.java
@@ -66,11 +66,15 @@ public class RegularContextFactory implements GLSurfaceView.EGLContextFactory {
 
 		GLUtils.checkEglError(TAG, "Before eglCreateContext", egl);
 		EGLContext context;
+		int[] debug_attrib_list = { EGL_CONTEXT_CLIENT_VERSION, 3, _EGL_CONTEXT_FLAGS_KHR, _EGL_CONTEXT_OPENGL_DEBUG_BIT_KHR, EGL10.EGL_NONE };
+		int[] attrib_list = { EGL_CONTEXT_CLIENT_VERSION, 3, EGL10.EGL_NONE };
 		if (mUseDebugOpengl) {
-			int[] attrib_list = { EGL_CONTEXT_CLIENT_VERSION, 3, _EGL_CONTEXT_FLAGS_KHR, _EGL_CONTEXT_OPENGL_DEBUG_BIT_KHR, EGL10.EGL_NONE };
-			context = egl.eglCreateContext(display, eglConfig, EGL10.EGL_NO_CONTEXT, attrib_list);
+			context = egl.eglCreateContext(display, eglConfig, EGL10.EGL_NO_CONTEXT, debug_attrib_list);
+			if (context == null || context == EGL10.EGL_NO_CONTEXT) {
+				Log.w(TAG, "creating 'OpenGL Debug' context failed");
+				context = egl.eglCreateContext(display, eglConfig, EGL10.EGL_NO_CONTEXT, attrib_list);
+			}
 		} else {
-			int[] attrib_list = { EGL_CONTEXT_CLIENT_VERSION, 3, EGL10.EGL_NONE };
 			context = egl.eglCreateContext(display, eglConfig, EGL10.EGL_NO_CONTEXT, attrib_list);
 		}
 		GLUtils.checkEglError(TAG, "After eglCreateContext", egl);

